TM1py/Services/CellService.py

```diff
@@ -991,6 +1008,138 @@ def write_through_unbound_process(self, cube_name: str, cellset_as_dict: Dict, i
 
+    @require_admin
+    @manage_transaction_log
+    @require_pandas
+    def write_through_blob(self, cube_name: str, cellset_as_dict: dict, increment: bool = False,
+                           sandbox_name: str = None, skip_non_updateable: bool = False,
+                           **kwargs):
+        """
+        Writes data back to TM1 via an unbound TI process having an uploaded CSV as data source
+        :param cube_name: str
+        :param cellset_as_dict:
+        :param increment: increment or update cell values
+        :param sandbox_name: str
+        :param skip_non_updateable skip cells that are not updateable (e.g. rule derived or consolidated)
+        :param kwargs:
+        :return: Success: bool, Messages: list, ChangeSet: None
+        """
+        from TM1py.Objects.Application import FolderApplication
+        from TM1py import ApplicationService
+        from TM1py import ProcessService
+        import os
+        import random
+        application_service = ApplicationService(self._rest)
+        process_service = ProcessService(self._rest)
+        cube_service = self.get_cube_service()
+
+        # Transform the cellset dict into a pandas dataframe and export to CSV
+
+        # Generate a random key to name uniquely the data extract file
+        random_key = random.randint(1000, 9999).__str__()
+        # Transform the cellset into a dictionary and export to CSV
+        df = build_pandas_dataframe_from_cellset(cellset_as_dict, multiindex=False)
+        # Define the path and file name
+        export_path = '.\\'
+        export_filename = 'data-source-extract-' + random_key + '.csv'
+        export_path_to_file = export_path + export_filename
+        # export cell dataframe to csv without indexes adn with all values with doublequotes
+        df.to_csv(export_path_to_file, index=False, sep=',', quoting=csv.QUOTE_ALL)
+
+        # Upload CSV to TM1 server using ApplicationService:
+        # Create a folder TM1py
+        tm1py_folder = FolderApplication(path='', name='TM1py')
+        if not application_service.exists(path='', application_type='FOLDER', name=tm1py_folder.name):
+            application_service.create(tm1py_folder)
+        # Update or create the CSV file inside the TM1py folder
+        application_service.update_or_create_document_from_file(path=tm1py_folder.name, name=export_filename,
+                                                                path_to_file=export_path_to_file)
+        # Retrieve the file ID to be used as data source in the TI process. Default folder is .\Externals
+        blob_filename = application_service.get_document(path=tm1py_folder.name, name=export_filename).file_id
+
+        # Create a TI processto load CSV  data into the cube
+        dataload_process_name = 'cube.' + cube_name + '.load.fromfile.' + random_key + 'test'
+        dataload_process = Process(name=dataload_process_name,
```

Comment:
Let's move the process creation to its own method. Perhaps a utility like "create_csv_load_process"